tests/Unit/Controllers/Api/ApiVoucherControllerTest.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace Tests\Unit\Controllers\Api;
+
+use App\Voucher;
+use App\StateToken;
+use App\Trader;
+use App\User;
+use App\VoucherState;
+use Auth;
+use Carbon\Carbon;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class ApiVoucherControllerTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    protected $trader;
+    protected $user;
+    protected $vouchers;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        // Create a Trader
+        $this->trader = factory(Trader::class)->create();
+
+        // Create a user on that trader
+        $this->user = factory(User::class)->create();
+        $this->user->traders()->sync([$this->trader->id]);
+
+        Auth::login($this->user);
+
+        // Create some vouchers at dispatched state
+        $this->vouchers = factory(Voucher::class, 'requested', 10)->create();
+        $this->vouchers->each(function ($voucher) {
+            $voucher->applyTransition('order');
+            $voucher->applyTransition('print');
+            $voucher->applyTransition('dispatch');
+        });
+    }
+
+    /** @test */
+    public function testItTidiesOldTokensOnConfirmTransitions()
```

Comment:
I feel like this test could be 2 or 3 separate tests - but probably not worth the extra work now to split out.